### ISSUE TITLE
ParentFolder should not submit

### ISSUE THREAD

**Issue by gmkll** [simogeo/Filemanager#504]
Except DOM-Reloading with XHR-Request a Full-Page-Reload is triggered with current code

`   $('#parentfolder').click(function() {  getFolderInfo(currentpath);   } );`

Returnig false is required to avoid this...

`   $('#parentfolder').click(function() {  
  getFolderInfo(currentpath);  
  return false;     
 } );`

**Comment by servocoder** [comment]
You are right, the bug exists. This repo isn't maintained currently. But this bug is already fixed at my repo: [Rich Filemanager](https://github.com/servocoder/RichFilemanager), which has a lot of fixes, improvements and new features. Here is the [changelog](https://github.com/servocoder/RichFilemanager/blob/master/changelog).

**Comment by gmkll** [comment]
Hi, thanks for pointing me to Rich FileManager! I just finished migrating an old Simogeo-Filemanager. In my forked scripts/filemanager.js code I started using async ajax calls, e.g. loading config files and elsewhere, but not completely up to now) and cleaned up a little bit the global space... , Sync calls are really deprecated! I am using JSP connector, which I updated as well, but not included in my git repo because of specific changes and cleaning work I have to do, which I have not done yet ..

**Comment by servocoder** [comment]
Cool! Maybe you would like to become a maintainer of JSP connector in my repo. The [API](https://github.com/servocoder/RichFilemanager/wiki/API) is changed a bit due to client-side modifications. Check the [roadmap](https://github.com/simogeo/Filemanager/issues/456) to be aware of planned and implemented features. By the way a developer for Java backend has joined recently.

**Comment by gmkll** [comment]
Hi, I uploaded the Java changes into my repo and made a Pull request #505, where you can cherry pick, what you want. Be aware, that FileManager.java requires Java 7 as it uses java.nio.

**Comment by servocoder** [comment]
Could you make the same PR to [RichFilemanager](https://github.com/servocoder/RichFilemanager) repo?

**Comment by servocoder** [comment]
But anyway it will require some refactoring and polishing, because of a number of changes in my repo. Will you be able to test and adjust Java connector after I will apply it to my repo.

**Comment by servocoder** [comment]
Since I'm not familiar with Java I need your help to make the Java connector work. And keep in mind that this repo is weakly supported, so if you want to help others to get Java connector worked together with a new features, then I would suggest you to migrate to RichFilemanager repo. It was agreed with simogeo - maintainer of the current repo.

**Comment by gmkll** [comment]
I´ll give it a try. I probably have to merge into RichFileManager filemanager.js, Many things seem to work, What´s changed, is that Preview property is now Thumbnail, although PReview seems to exist still. Rename and Select are ok, Upload not yet...

**Comment by servocoder** [comment]
Yeah, there were a lot of changes. Concerning to Preview and Thumbnail an [issue](https://github.com/servocoder/RichFilemanager/issues/18) was found recently. I'm going to handle it, perhaps some modifications will be made. You'd better suspend on that. But the other stuff is good.

**Comment by gmkll** [comment]
I have forked RichFileManager repo and managed to apply the changes I did (as in PullRequest #505) and fixed the download support in Java connector. I also added an abstract class (to be reused in both Filemanagers), and just moved only the methods specific to the new RichFM (which is referenced in filemanager.jsp). Unfortunately I see for the moment no easy way to implement the upload features, so  I´ll stick with old simogeo FM, where I updated my changes accordingly (yet in forks). Though I could provide pull requests for both..

**Comment by servocoder** [comment]
That is great! I will apply your PR to the current repo. Could I expect you to implement the upload features for RichFilemanager and create PR soon?

**Comment by gmkll** [comment]
I updated and finshed all, last step was moving data holder variable to local scope again. I merged the last changes from your repo, but nevertheless some major reordering especially at the beginning and at the end of filemanager.js could not be avoided and should be reviewed. Check the [PR] (https://github.com/servocoder/RichFilemanager/pull/21).

**Comment by gmkll** [comment]
I´ll may work on it soon (it seems textarea html content type is not required for replace?). In the current PR I have added an old check for fileRoot, which I should remove, as it seems not required any more...

**Comment by gmkll** [comment]
I tested upload and replace, which seemt to work now, although fileTree refresh seems not .. another TODO.  I committed an [update ](https://github.com/servocoder/RichFilemanager/pull/21/commits/d42d036213bea9ecfe30c1ec9e04b6fa2bb1d0c0)

**Comment by simogeo** [comment]
[RichFilemanager](https://github.com/servocoder/RichFilemanager) is now the reference ! Happy to see that more and more people are getting involved.
